MediaBrowser.Controller/Playlists/Playlist.cs

```diff
@@ -43,6 +44,16 @@ public Playlist()
 
         public static bool IsPlaylistFile(string path)
         {
+            //When a directory contains a `.`, calling "HasExtension" will return true, even if that location is a directory that exists.
+            //This kills the PlaylistXmlSaver, because instead of saving in "config/data/playlists/MyList2.0/playlist.xml",
+            //It saves the information in "config/data/playlists/MyList2.xml". So we just need to see if it's actually a real directory first.
+            //Lucky for us, when a new playlist is created, the directory on the drive is created first, then the Playlist object is created.
+            //And if there's not a directory there, then we can just check if it has an extension.
+            if (new System.IO.DirectoryInfo(path).Exists)
+            { //This is a directory, and therefore definitely not a playlist file.
+                return false;
+            }
+            //Well, there's no directory there, so if it /looks/ like a file path, then it probably is.
             return System.IO.Path.HasExtension(path);
```

Comment:
Also add `using System.IO;` further up top.
```suggestion
            // "Path.HasExtension" returns true if the name contains a '.' (dot). Make sure it's not a directory before checking the extension.
            return !Directory.Exists(path) && Path.HasExtension(path);
```

Reply:
Will do, but I think I might have missed part of the issue, because while this seems to completely explain the issue, the last playlist I tried to create still had the problem.